juju/api_whitebox_test.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package juju
+
+import (
+	gc "gopkg.in/check.v1"
+	jc "github.com/juju/testing/checkers"
+)
+
+type APIHelperSuite struct {
+}
+
+var _ = gc.Suite(&APIHelperSuite{})
+
+var moveToFrontTests = []struct {
+	item   string
+	items  []string
+	expect []string
+}{{
+	item:   "x",
+	items:  []string{"y", "x"},
+	expect: []string{"x", "y"},
+}, {
+	item:   "z",
+	items:  []string{"y", "x"},
+	expect: []string{"y", "x"},
+}, {
+	item:   "y",
+	items:  []string{"y", "x"},
+	expect: []string{"y", "x"},
+}, {
+	item:   "x",
+	items:  []string{"y", "x", "z"},
+	expect: []string{"x", "y", "z"},
+}, {
+	item:   "d",
+	items:  []string{"a", "b", "c", "d", "e", "f"},
+	expect: []string{"d", "a", "b", "c", "e", "f"},
+}}
+
+func (s *APIHelperSuite) TestMoveToFront(c *gc.C) {
```

Comment:
the move-to-front tests have not been altered, just moved to a package test rather than an fake exported function for testing test.